From 2af45c06d68dfa8dabb4c71f93b6c574daed3d06 Mon Sep 17 00:00:00 2001
From: Michael-CY Lee <michael-cy.lee@mediatek.com>
Date: Tue, 9 Sep 2025 11:19:35 +0800
Subject: [PATCH 230/254] mtk: hostapd: move hostapd_interfaces_get_hapd to
 hostapd.c

The function will be used to extend hostapd_interface_init_bss() for
link addition in a following patch

Signed-off-by: Michael-CY Lee <michael-cy.lee@mediatek.com>
---
 hostapd/ctrl_iface.c | 22 ----------------------
 src/ap/hostapd.c     | 21 +++++++++++++++++++++
 src/ap/hostapd.h     |  3 +++
 3 files changed, 24 insertions(+), 22 deletions(-)

diff --git a/hostapd/ctrl_iface.c b/hostapd/ctrl_iface.c
index cb00537db..ba95fcb81 100644
--- a/hostapd/ctrl_iface.c
+++ b/hostapd/ctrl_iface.c
@@ -8267,28 +8267,6 @@ hostapd_global_ctrl_iface_fst_detach(struct hapd_interfaces *interfaces,
 #endif /* CONFIG_FST */
 
 
-static struct hostapd_data *
-hostapd_interfaces_get_hapd(struct hapd_interfaces *interfaces,
-			    const char *ifname)
-{
-	size_t i, j;
-
-	for (i = 0; i < interfaces->count; i++) {
-		struct hostapd_iface *iface = interfaces->iface[i];
-
-		for (j = 0; j < iface->num_bss; j++) {
-			struct hostapd_data *hapd;
-
-			hapd = iface->bss[j];
-			if (os_strcmp(ifname, hapd->conf->iface) == 0)
-				return hapd;
-		}
-	}
-
-	return NULL;
-}
-
-
 static int hostapd_ctrl_iface_dup_param(struct hostapd_data *src_hapd,
 					struct hostapd_data *dst_hapd,
 					const char *param)
diff --git a/src/ap/hostapd.c b/src/ap/hostapd.c
index 020f752bd..39ca9277b 100644
--- a/src/ap/hostapd.c
+++ b/src/ap/hostapd.c
@@ -6023,3 +6023,24 @@ hostapd_epcs_get_entry(struct hapd_interfaces *ifaces, const u8 *addr)
 	return NULL;
 }
 
+
+struct hostapd_data *
+hostapd_interfaces_get_hapd(struct hapd_interfaces *interfaces,
+			    const char *ifname)
+{
+	size_t i, j;
+
+	for (i = 0; i < interfaces->count; i++) {
+		struct hostapd_iface *iface = interfaces->iface[i];
+
+		for (j = 0; j < iface->num_bss; j++) {
+			struct hostapd_data *hapd;
+
+			hapd = iface->bss[j];
+			if (os_strcmp(ifname, hapd->conf->iface) == 0)
+				return hapd;
+		}
+	}
+
+	return NULL;
+}
diff --git a/src/ap/hostapd.h b/src/ap/hostapd.h
index af4d88afb..f05e323a1 100644
--- a/src/ap/hostapd.h
+++ b/src/ap/hostapd.h
@@ -1139,4 +1139,7 @@ static inline int hostapd_twt_init(struct hostapd_data *hapd)
 	return 0;
 }
 
+struct hostapd_data *
+hostapd_interfaces_get_hapd(struct hapd_interfaces *interfaces,
+			    const char *ifname);
 #endif /* HOSTAPD_H */
-- 
2.45.2

